lib/ansible/modules/remote_management/hpilo/hpilo_boot.py

```diff
@@ -0,0 +1,208 @@
+    media: cdrom
+    image: http://some-web-server/iso/boot.iso
+  when: cmdb_hwmodel.startswith('HP ')
+  delegate_to: localhost
+
+- name: Power off a server
+  hpilo_boot:
+    host: YOUR_ILO_HOST
+    login: YOUR_ILO_LOGIN
+    password: YOUR_ILO_PASSWORD
+    state: poweroff
+  delegate_to: localhost
+'''
+
+RETURN = '''
+# Default return values
+'''
+
+
+import time
+import warnings
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    import hpilo
+    HAS_HPILO = True
+except ImportError:
+    HAS_HPILO = False
+
+
+# Suppress warnings from hpilo
+warnings.simplefilter('ignore')
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            host = dict(required=True, type='str'),
+            login = dict(default='Administrator', type='str'),
+            password = dict(default='admin', type='str', no_log=True),
+            media = dict(default=None, type='str', choices=['cdrom', 'floppy', 'rbsu', 'hdd', 'network', 'normal', 'usb']),
+            image = dict(default=None, type='str'),
+            state = dict(default='boot_once', type='str', choices=['boot_always', 'boot_once', 'connect', 'disconnect', 'no_boot', 'poweroff']),
+            force = dict(default=False, type='bool'),
+        )
+    )
+
+    if not HAS_HPILO:
+        module.fail_json(msg='The hpilo python module is required')
+
+    host = module.params['host']
+    login = module.params['login']
+    password = module.params['password']
+    media = module.params['media']
+    image = module.params['image']
+    state = module.params['state']
+    force = module.params['force']
+
+    ilo = hpilo.Ilo(host, login=login, password=password)
+    changed = False
+    status = {}
+    power_status = 'UNKNOWN'
+
+    if media and state in ('boot_always', 'boot_once', 'connect', 'disconnect', 'no_boot'):
+
+        # Workaround for: Error communicating with iLO: Problem manipulating EV
+        try:
+            ilo.set_one_time_boot(media)
+        except hpilo.IloError:
+            time.sleep(60)
+            ilo.set_one_time_boot(media)
+
+        # TODO: Verify if image URL exists/works
+        if image:
+            ilo.insert_virtual_media(media, image)
+            changed = True
+
+        if media == 'cdrom':
+            ilo.set_vm_status('cdrom', state, True)
+            status = ilo.get_vm_status()
+            changed = True
+        elif media in ('floppy', 'usb'):
+            ilo.set_vf_status(state, True)
+            status = ilo.get_vf_status()
+            changed = True
+
+    # Only perform a boot when state is boot_once or boot_always, or in case we want to force a reboot
+    if state in ('boot_once', 'boot_always') or force:
+
+        power_status = ilo.get_host_power_status()
+
+        if not force and power_status == 'ON':
+            module.fail_json(msg='HP iLO (%s) reports that the server is already powered on !' % host)
+
+        if power_status == 'ON':
+            #ilo.cold_boot_server()
```

Comment:
Is that really a problem ? There are two ways of doing this through the existing API.
The commented code shows the other alternative (which I think was once active).

I don't think this is a problem, instead it may lead to people tinker with it if they have issues or expected different behaviour on their specific model. (Because there is a problem that various models exist and we cannot possibly test against all of them)